### PR TITLE
Show service status in system tray

### PR DESCRIPTION
**Pull request by tadaskay**
Ref: https://github.com/mysteriumnetwork/mysterium-vpn/issues/465

- Added provider service status menu item in the system tray
- Renamed `Status` to `VPN` now that we have 2 statuses (VPN connection and Provider service)

Screenshot 2019-03-27 at 11 05 08

### CODE REVIEW

File: src/app/communication/main-communication.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import type {
   ConnectionStatusChangeDTO,
   CountriesDTO,
   CurrentIdentityChangeDTO,
-  FavoriteProviderDTO,
+  FavoriteProviderDTO, ProviderServiceStatusChangeDTO,
```
Comment (donce) [inline src/app/communication/main-communication.js]:
```suggestion
  FavoriteProviderDTO,
  ProviderServiceStatusChangeDTO,
```

File: src/app/communication/dto.js [hunk 2]
```diff
@@ -61,3 +62,7 @@ export type MetricDto = {
   metric: Metric,
   value: mixed
 }
+
+export type ProviderServiceStatusChangeDTO = {
```
Comment (donce) [inline src/app/communication/dto.js]: I don't think we really need this DTO - we can use `ServiceStatus` directly.

File: src/app/communication/message-transport.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ import type {
   ConnectionStatusChangeDTO,
   CountriesDTO,
   CurrentIdentityChangeDTO,
-  FavoriteProviderDTO,
+  FavoriteProviderDTO, ProviderServiceStatusChangeDTO,
```
Comment (donce) [inline src/app/communication/message-transport.js]:
```suggestion
  FavoriteProviderDTO,
  ProviderServiceStatusChangeDTO,
```

Comment (tadaskay) [reply src/app/communication/message-transport.js]: Fixed, thanks!

File: src/app/communication/dto.js (hunk 2 above)
Comment (tadaskay) [reply src/app/communication/dto.js]: Removed, thanks!

File: src/app/communication/main-communication.js (hunk 1 above)
Comment (tadaskay) [reply src/app/communication/main-communication.js]: 👍

File: test/unit/main/tray/menu-builder.spec.js [hunk 4]
```diff
@@ -85,44 +85,44 @@ describe('tray', () => {
...
       it('renders menu items without disconnect when not connected', () => {
         const items = builder.updateConnectionStatus(ConnectionStatus.NOT_CONNECTED).build()
-        expect(items[1].type).to.equal(separator)
-        expect(items[2].label).to.equal(translations.connect)
-        expect(items[3].type).to.equal(separator)
-        expect(items[4].label).to.equal(translations.showWindow)
-        expect(items[5].label).to.equal(translations.toggleDeveloperTools)
-        expect(items[6].type).to.equal(separator)
-        expect(items[7].label).to.equal(translations.quit)
+        expect(items[2].type).to.equal(separator)
```
Comment (donce) [inline test/unit/main/tray/menu-builder.spec.js]: Maybe it makes sense to test 1st element as well?

Comment (tadaskay) [reply test/unit/main/tray/menu-builder.spec.js]: Sorry, I missed this coment. What do you think about testing independently from absolute order?

Comment (donce) [reply test/unit/main/tray/menu-builder.spec.js]: That's alright, that was just a small comment. What do you mean by testing independently from absolute order?

Comment (tadaskay) [reply test/unit/main/tray/menu-builder.spec.js]: If there's no need to be so strict about the order of the elements then you could test that menu contains such item, not necessarily at the index `2`. That way you ensure that the functionality is there and at the same time make the test future-proof by not failing when adding new items to the list.

**Pull request merged**